kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -652,6 +652,11 @@ private void updateBaseRecordTimestampOfEncodedPartition(SinkRecord record,
   }
 
   protected void commitFiles() {
```

Comment:
who else invoke this? because from `commitOnTimeIfNoData()` there's already validation as you stated if it's empty

Reply:
Our new "flush on termination/revoke" code:
https://github.com/logzio/kafka-connect-storage-cloud/blob/f6700eadcf68a3b8d9efbef0e1f1253c48c4c842/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java#L377